countlyCommon/countlyCommon/Helpers/TimeHelper.cs

```diff
@@ -21,12 +21,59 @@ THE SOFTWARE.
 */
 
 using System;
+using System.Globalization;
 
 namespace CountlySDK.Helpers
 {
     internal class TimeHelper
     {
 
+        internal class TimeInstant
+        {
+            public int Dow { get; private set; }
+            public int Hour { get; private set; }
+            public int Timezone { get; private set; }
+            public long Timestamp { get; private set; }
+
+            
+
+            internal TimeInstant()
+            {
+                DateTime dateTime = DateTime.Now;
+
+                Dow = (int)dateTime.DayOfWeek;
+                Hour = dateTime.TimeOfDay.Hours;
+
+                TimeSpan ts = dateTime.Subtract(new DateTime(1970, 1, 1));
+                Timestamp = (long)ts.TotalMilliseconds;
+
+                Timezone = (int)TimeZoneInfo.Local.GetUtcOffset(dateTime).TotalMinutes;
+            }
+
+            internal TimeInstant(long timestampInMillis, int hour, int dow)
```

Comment:
This constructor needs all 4 fields. The timezone is not set bellow and you are already introducing a bug.